{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Unpacking Cellular Clarity Dataset to Preprocess and Prepare the Dataset"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Teague McCracken 2/24/25"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Note: create a new environment if sequence alignment tools are needed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import socket\n",
    "\n",
    "# Define paths\n",
    "# System Switch - @Mohsen, add your systems if you want to...\n",
    "hostname = socket.gethostname()\n",
    "if hostname in ['Teagues-MacBook-Pro.local', 'Teagues-MBP.attlocal.net']:\n",
    "    sys_id = 1\n",
    "else:\n",
    "    sys_id = 0\n",
    "\n",
    "# Setting programs path based on which system was identified by the system switch\n",
    "if sys_id == 1: \n",
    "    programs_dir = '/Users/teaguemcc/Programs'\n",
    "elif sys_id == 0:\n",
    "    programs_dir = '/home/temccrac/Programs'\n",
    "\n",
    "git_path = os.path.join(programs_dir, 'git_clones/ECE759_Project')\n",
    "git_data_path = os.path.join(git_path, 'cellular_clarity')\n",
    "data_path = os.path.join(programs_dir, 'data/cellular_clarity')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "epidermis_RPKM = pd.read_excel(os.path.join(git_data_path, 'GSE228011_epidermis_RPKM.xlsx'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['AGI', 'Alias', 'Description', 'A1', 'A2', 'A3', 'Mean  A', 'B+1',\n",
       "       'B+2', 'B+3', 'Mean B+', 'C+1', 'C+2', 'C+3', 'Mean C+', 'D+1', 'D+2',\n",
       "       'D+3', 'Mean D+', 'E+1', 'E+2', 'E+3', 'Mean E+', 'F+1', 'F+2', 'F+3',\n",
       "       'Mean F+', 'G+1', 'G+3', 'G+4', 'Mean G+', 'B-1', 'B-2', 'B-3',\n",
       "       'Mean B-', 'C-1', 'C-2', 'C-3', 'Mean C-', 'D-1', 'D-2', 'D-3',\n",
       "       'Mean D-', 'E-1', 'E-2', 'E-3', 'Mean E-', 'F-1', 'F-2', 'F-3',\n",
       "       'Mean F-', 'G-1', 'G-2', 'G-4', 'Mean G-'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "epidermis_RPKM.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "AGI",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "Alias",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "Description",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "A1",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "A2",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "A3",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "Mean  A",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "B+1",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "B+2",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "B+3",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "Mean B+",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "C+1",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "C+2",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "C+3",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "Mean C+",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "D+1",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "D+2",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "D+3",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "Mean D+",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "E+1",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "E+2",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "E+3",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "Mean E+",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "F+1",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "F+2",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "F+3",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "Mean F+",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "G+1",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "G+3",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "G+4",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "Mean G+",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "B-1",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "B-2",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "B-3",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "Mean B-",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "C-1",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "C-2",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "C-3",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "Mean C-",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "D-1",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "D-2",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "D-3",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "Mean D-",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "E-1",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "E-2",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "E-3",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "Mean E-",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "F-1",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "F-2",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "F-3",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "Mean F-",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "G-1",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "G-2",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "G-4",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "Mean G-",
         "rawType": "float64",
         "type": "float"
        }
       ],
       "conversionMethod": "pd.DataFrame",
       "ref": "11445430-9b28-410c-b47a-cc6c166d85ec",
       "rows": [
        [
         "0",
         "AT1G01010",
         "NAC001",
         "NAC domain containing protein 1",
         "9.39528474046342",
         "10.5653482584335",
         "8.50767503836295",
         "9.48943601241996",
         "6.16853495008346",
         "6.55810253874885",
         "5.52867636281331",
         "6.08510461721521",
         "4.85522303524273",
         "6.35460961144699",
         "9.44436885965439",
         "6.88473383544804",
         "3.60697958181321",
         "6.56261684723072",
         "9.17513593544443",
         "6.44824412149612",
         "6.93915371737636",
         "7.91022601883701",
         "6.72098756318581",
         "7.19012243313306",
         "4.15022177991492",
         "5.85298548729016",
         "7.51241083762002",
         "5.83853936827503",
         "10.0868168737761",
         "13.6350798582264",
         "7.30396362169213",
         "10.3419534512315",
         "7.94488409612198",
         "9.951783937325",
         "8.04138042491206",
         "8.64601615278635",
         "8.39107876806819",
         "10.2131838201501",
         "9.00896674766456",
         "9.20440977862762",
         "8.88792153643878",
         "7.1020446269293",
         "7.20182071988809",
         "7.73059562775206",
         "5.91103582758453",
         "8.13174138786709",
         "6.10694871809203",
         "6.71657531118122",
         "7.0563391588367",
         "9.98730427708085",
         "8.87677706506055",
         "8.6401401669927",
         "13.7428121329727",
         "12.4947206257942",
         "8.31729014160477",
         "11.5182743001239"
        ],
        [
         "1",
         "AT1G01020",
         "ARV1",
         null,
         "12.1018141831472",
         "11.0640935438218",
         "10.2102124027034",
         "11.1253733765575",
         "12.7768920225737",
         "11.6173540379141",
         "13.7987870832293",
         "12.7310110479057",
         "14.6116232106474",
         "11.2991099806559",
         "13.0713085047425",
         "12.9940138986819",
         "11.0808457618518",
         "10.5501398852024",
         "9.53042894354092",
         "10.387138196865",
         "9.96844854215116",
         "11.304440353736",
         "13.4879894718898",
         "11.5869594559257",
         "11.3429506489363",
         "12.1725178128674",
         "10.2002747914675",
         "11.2385810844237",
         "10.8027501368074",
         "11.9955863072234",
         "11.3812287109668",
         "11.3931883849992",
         "10.3549268601716",
         "10.2455167066138",
         "12.0165285927108",
         "10.8723240531654",
         "9.67111900061072",
         "10.6510035513535",
         "12.4838299739969",
         "10.9353175086537",
         "10.0095551705244",
         "10.7696094027108",
         "8.5954738187912",
         "9.79154613067547",
         "9.02584624115318",
         "9.92224525125302",
         "8.7534347679084",
         "9.23384208677153",
         "8.69723035974172",
         "9.19053630695475",
         "11.1325982344339",
         "9.67345496704346",
         "12.5973411936354",
         "10.8982512485789",
         "12.8317804531489",
         "12.1091242984544"
        ],
        [
         "2",
         "AT1G01030",
         "NGA3",
         "NGATHA3",
         "1.73996445590576",
         "2.35353554659337",
         "1.6299587673697",
         "1.90781958995628",
         "1.06651169535788",
         "1.79277509061727",
         "1.59408671298414",
         "1.48445783298643",
         "1.70085430286669",
         "1.25127804305891",
         "2.00568640340904",
         "1.65260624977821",
         "1.62088222317641",
         "2.08685770509095",
         "1.49013279455264",
         "1.73262424094",
         "1.13791051244347",
         "1.33254109355976",
         "1.26326555391082",
         "1.24457238663802",
         "1.40839156263958",
         "1.28389682935098",
         "1.72026210362889",
         "1.47085016520648",
         "2.05066863186535",
         "3.00360022472628",
         "1.92971390732274",
         "2.32799425463812",
         "1.59728285831119",
         "1.83109513037527",
         "1.76938982831583",
         "1.7325892723341",
         "1.82215853545292",
         "1.73756011725742",
         "2.24756996270252",
         "1.93576287180429",
         "1.5979258377962",
         "2.67329893444043",
         "2.07988190340246",
         "2.11703555854636",
         "1.24214353452587",
         "2.03447958687832",
         "1.77306476033446",
         "1.68322929391288",
         "2.10577980636602",
         "1.86308270420914",
         "2.29298011385387",
         "2.08728087480968",
         "2.67774363133455",
         "3.12888450773056",
         "1.72971189677152",
         "2.51211334527888"
        ],
        [
         "3",
         "AT1G01040",
         "ASU1, ATDCL1, CAF, DCL1, EMB60, EMB76, SIN1, SUS1",
         "ABNORMAL SUSPENSOR 1, DICER-LIKE 1, CARPEL FACTORY, dicer-like 1, EMBRYO DEFECTIVE 60, EMBRYO DEFECTIVE 76, SHORT INTEGUMENTS 1, SUSPENSOR 1",
         "0.913234154908988",
         "1.03552051684939",
         "1.36535760993892",
         "1.1047040938991",
         "0.89337834091079",
         "0.790886577419624",
         "0.757958036796067",
         "0.814074318375494",
         "1.03617762934476",
         "0.779616029313786",
         "0.737344049862147",
         "0.851045902840231",
         "1.48118674535549",
         "1.47132005262042",
         "1.50855657564747",
         "1.48702112454113",
         "0.912140241279189",
         "1.12028055032275",
         "1.28632163636261",
         "1.10624747598818",
         "1.15592525184928",
         "1.47684709784729",
         "1.5150450073203",
         "1.38260578567229",
         "0.883930226625561",
         "0.503715488113808",
         "0.868136280463415",
         "0.751927331734261",
         "1.18932104949405",
         "0.689862077736143",
         "1.15076328322799",
         "1.00998213681939",
         "1.39716637480432",
         "1.1688036805704",
         "1.36924233416959",
         "1.31173746318144",
         "1.43703093412299",
         "1.08236197316036",
         "1.50226374288003",
         "1.34055221672113",
         "1.43778014813444",
         "1.18778279366637",
         "1.04509854866316",
         "1.22355383015466",
         "1.9637661944216",
         "1.63157516559249",
         "0.965135508572736",
         "1.52015895619561",
         "0.495218645785379",
         "0.65982048270695",
         "0.549524212107372",
         "0.5681877801999"
        ],
        [
         "4",
         "AT1G01046",
         "MIR838A",
         "microRNA838A",
         "0.246349482608731",
         "0.0",
         "0.0",
         "0.0821164942029103",
         "0.306718716735621",
         "0.0",
         "0.0",
         "0.102239572245207",
         "0.0",
         "0.523426585864565",
         "0.0",
         "0.174475528621522",
         "0.210095981162895",
         "0.0",
         "0.0",
         "0.070031993720965",
         "0.1837206141372",
         "0.0",
         "0.166081496218089",
         "0.116600703451763",
         "0.180017842648175",
         "0.0",
         "0.0",
         "0.0600059475493917",
         "0.0",
         "0.0",
         "0.273214789554056",
         "0.0910715965180187",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.713246839738182",
         "0.237748946579394",
         "0.350132128019526",
         "0.152808067636122",
         "0.0",
         "0.167646731885216",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.0",
         "0.293369083143105",
         "0.0",
         "0.0",
         "0.097789694381035"
        ]
       ],
       "shape": {
        "columns": 55,
        "rows": 5
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AGI</th>\n",
       "      <th>Alias</th>\n",
       "      <th>Description</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>A3</th>\n",
       "      <th>Mean  A</th>\n",
       "      <th>B+1</th>\n",
       "      <th>B+2</th>\n",
       "      <th>B+3</th>\n",
       "      <th>...</th>\n",
       "      <th>E-3</th>\n",
       "      <th>Mean E-</th>\n",
       "      <th>F-1</th>\n",
       "      <th>F-2</th>\n",
       "      <th>F-3</th>\n",
       "      <th>Mean F-</th>\n",
       "      <th>G-1</th>\n",
       "      <th>G-2</th>\n",
       "      <th>G-4</th>\n",
       "      <th>Mean G-</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AT1G01010</td>\n",
       "      <td>NAC001</td>\n",
       "      <td>NAC domain containing protein 1</td>\n",
       "      <td>9.395285</td>\n",
       "      <td>10.565348</td>\n",
       "      <td>8.507675</td>\n",
       "      <td>9.489436</td>\n",
       "      <td>6.168535</td>\n",
       "      <td>6.558103</td>\n",
       "      <td>5.528676</td>\n",
       "      <td>...</td>\n",
       "      <td>6.106949</td>\n",
       "      <td>6.716575</td>\n",
       "      <td>7.056339</td>\n",
       "      <td>9.987304</td>\n",
       "      <td>8.876777</td>\n",
       "      <td>8.640140</td>\n",
       "      <td>13.742812</td>\n",
       "      <td>12.494721</td>\n",
       "      <td>8.317290</td>\n",
       "      <td>11.518274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AT1G01020</td>\n",
       "      <td>ARV1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.101814</td>\n",
       "      <td>11.064094</td>\n",
       "      <td>10.210212</td>\n",
       "      <td>11.125373</td>\n",
       "      <td>12.776892</td>\n",
       "      <td>11.617354</td>\n",
       "      <td>13.798787</td>\n",
       "      <td>...</td>\n",
       "      <td>8.753435</td>\n",
       "      <td>9.233842</td>\n",
       "      <td>8.697230</td>\n",
       "      <td>9.190536</td>\n",
       "      <td>11.132598</td>\n",
       "      <td>9.673455</td>\n",
       "      <td>12.597341</td>\n",
       "      <td>10.898251</td>\n",
       "      <td>12.831780</td>\n",
       "      <td>12.109124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AT1G01030</td>\n",
       "      <td>NGA3</td>\n",
       "      <td>NGATHA3</td>\n",
       "      <td>1.739964</td>\n",
       "      <td>2.353536</td>\n",
       "      <td>1.629959</td>\n",
       "      <td>1.907820</td>\n",
       "      <td>1.066512</td>\n",
       "      <td>1.792775</td>\n",
       "      <td>1.594087</td>\n",
       "      <td>...</td>\n",
       "      <td>1.773065</td>\n",
       "      <td>1.683229</td>\n",
       "      <td>2.105780</td>\n",
       "      <td>1.863083</td>\n",
       "      <td>2.292980</td>\n",
       "      <td>2.087281</td>\n",
       "      <td>2.677744</td>\n",
       "      <td>3.128885</td>\n",
       "      <td>1.729712</td>\n",
       "      <td>2.512113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AT1G01040</td>\n",
       "      <td>ASU1, ATDCL1, CAF, DCL1, EMB60, EMB76, SIN1, SUS1</td>\n",
       "      <td>ABNORMAL SUSPENSOR 1, DICER-LIKE 1, CARPEL FAC...</td>\n",
       "      <td>0.913234</td>\n",
       "      <td>1.035521</td>\n",
       "      <td>1.365358</td>\n",
       "      <td>1.104704</td>\n",
       "      <td>0.893378</td>\n",
       "      <td>0.790887</td>\n",
       "      <td>0.757958</td>\n",
       "      <td>...</td>\n",
       "      <td>1.045099</td>\n",
       "      <td>1.223554</td>\n",
       "      <td>1.963766</td>\n",
       "      <td>1.631575</td>\n",
       "      <td>0.965136</td>\n",
       "      <td>1.520159</td>\n",
       "      <td>0.495219</td>\n",
       "      <td>0.659820</td>\n",
       "      <td>0.549524</td>\n",
       "      <td>0.568188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AT1G01046</td>\n",
       "      <td>MIR838A</td>\n",
       "      <td>microRNA838A</td>\n",
       "      <td>0.246349</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.082116</td>\n",
       "      <td>0.306719</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.293369</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.097790</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 55 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         AGI                                              Alias  \\\n",
       "0  AT1G01010                                             NAC001   \n",
       "1  AT1G01020                                               ARV1   \n",
       "2  AT1G01030                                               NGA3   \n",
       "3  AT1G01040  ASU1, ATDCL1, CAF, DCL1, EMB60, EMB76, SIN1, SUS1   \n",
       "4  AT1G01046                                            MIR838A   \n",
       "\n",
       "                                         Description         A1         A2  \\\n",
       "0                    NAC domain containing protein 1   9.395285  10.565348   \n",
       "1                                                NaN  12.101814  11.064094   \n",
       "2                                            NGATHA3   1.739964   2.353536   \n",
       "3  ABNORMAL SUSPENSOR 1, DICER-LIKE 1, CARPEL FAC...   0.913234   1.035521   \n",
       "4                                       microRNA838A   0.246349   0.000000   \n",
       "\n",
       "          A3    Mean  A        B+1        B+2        B+3  ...       E-3  \\\n",
       "0   8.507675   9.489436   6.168535   6.558103   5.528676  ...  6.106949   \n",
       "1  10.210212  11.125373  12.776892  11.617354  13.798787  ...  8.753435   \n",
       "2   1.629959   1.907820   1.066512   1.792775   1.594087  ...  1.773065   \n",
       "3   1.365358   1.104704   0.893378   0.790887   0.757958  ...  1.045099   \n",
       "4   0.000000   0.082116   0.306719   0.000000   0.000000  ...  0.000000   \n",
       "\n",
       "    Mean E-       F-1       F-2        F-3   Mean F-        G-1        G-2  \\\n",
       "0  6.716575  7.056339  9.987304   8.876777  8.640140  13.742812  12.494721   \n",
       "1  9.233842  8.697230  9.190536  11.132598  9.673455  12.597341  10.898251   \n",
       "2  1.683229  2.105780  1.863083   2.292980  2.087281   2.677744   3.128885   \n",
       "3  1.223554  1.963766  1.631575   0.965136  1.520159   0.495219   0.659820   \n",
       "4  0.000000  0.000000  0.000000   0.000000  0.000000   0.293369   0.000000   \n",
       "\n",
       "         G-4    Mean G-  \n",
       "0   8.317290  11.518274  \n",
       "1  12.831780  12.109124  \n",
       "2   1.729712   2.512113  \n",
       "3   0.549524   0.568188  \n",
       "4   0.000000   0.097790  \n",
       "\n",
       "[5 rows x 55 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "epidermis_RPKM.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['GeneID', 't0(Fe+)', 't6 (Fe+)', 't6 (Fe-)', 't12 (Fe+)', 't12 (Fe-)',\n",
       "       't18 (Fe+)', 't18 (Fe-)', 't24 (Fe+)', 't24 (Fe-)', 't30 (Fe+)',\n",
       "       't30 (Fe-)', 't36 (Fe+)', 't36(Fe-)', 'GeneName'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DE_genes_epidermis_RPKM = pd.read_csv(os.path.join(git_data_path, 'GSE228011_rpkm_DE_genes_epidermis.csv.gz'), compression = 'gzip')\n",
    "DE_genes_epidermis_RPKM.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "GeneID",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "t0(Fe+)",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "t6 (Fe+)",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "t6 (Fe-)",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "t12 (Fe+)",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "t12 (Fe-)",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "t18 (Fe+)",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "t18 (Fe-)",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "t24 (Fe+)",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "t24 (Fe-)",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "t30 (Fe+)",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "t30 (Fe-)",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "t36 (Fe+)",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "t36(Fe-)",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "GeneName",
         "rawType": "object",
         "type": "string"
        }
       ],
       "conversionMethod": "pd.DataFrame",
       "ref": "15ef3148-fb0e-4d74-8d5c-a78c703bfc40",
       "rows": [
        [
         "0",
         "AT1G03070",
         "671.3393063",
         "346.3705693",
         "534.5581171",
         "544.9180313",
         "548.0793347",
         "552.9255616",
         "531.8057139",
         "561.5208476",
         "646.8305014",
         "546.5319553",
         "511.3584942",
         "618.6424931",
         "466.0281198",
         "unknown"
        ],
        [
         "1",
         "AT1G03090",
         "56.48990899",
         "413.2322069",
         "115.2515911",
         "256.1826471",
         "92.07233697",
         "138.8238459",
         "76.02053554",
         "211.4053461",
         "51.50887988",
         "178.7866298",
         "43.95419705",
         "89.6474165",
         "55.00269224",
         "MCCA"
        ],
        [
         "2",
         "AT1G04610",
         "555.8427212",
         "334.2144281",
         "515.0207558",
         "389.253013",
         "504.6867178",
         "593.0030598",
         "737.0958892",
         "515.2494148",
         "602.2950807",
         "566.374468",
         "633.2931839",
         "449.8953631",
         "435.1543377",
         "YUC3"
        ],
        [
         "3",
         "AT1G04990",
         "442.0630759",
         "305.6875747",
         "386.7437032",
         "306.5713264",
         "360.5001732",
         "332.1707686",
         "373.9651895",
         "420.0780448",
         "374.7902218",
         "349.3001867",
         "377.9684627",
         "422.1885474",
         "398.5440919",
         "unknown"
        ],
        [
         "4",
         "AT1G05560",
         "820.8994473",
         "380.1289088",
         "695.3395147",
         "406.5960998",
         "845.0465693",
         "528.0742465",
         "632.5836435",
         "654.9783334",
         "966.961101",
         "577.2207485",
         "965.8815246",
         "917.4047497",
         "805.2413282",
         "UGT1, UGT75B1"
        ]
       ],
       "shape": {
        "columns": 15,
        "rows": 5
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneID</th>\n",
       "      <th>t0(Fe+)</th>\n",
       "      <th>t6 (Fe+)</th>\n",
       "      <th>t6 (Fe-)</th>\n",
       "      <th>t12 (Fe+)</th>\n",
       "      <th>t12 (Fe-)</th>\n",
       "      <th>t18 (Fe+)</th>\n",
       "      <th>t18 (Fe-)</th>\n",
       "      <th>t24 (Fe+)</th>\n",
       "      <th>t24 (Fe-)</th>\n",
       "      <th>t30 (Fe+)</th>\n",
       "      <th>t30 (Fe-)</th>\n",
       "      <th>t36 (Fe+)</th>\n",
       "      <th>t36(Fe-)</th>\n",
       "      <th>GeneName</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AT1G03070</td>\n",
       "      <td>671.339306</td>\n",
       "      <td>346.370569</td>\n",
       "      <td>534.558117</td>\n",
       "      <td>544.918031</td>\n",
       "      <td>548.079335</td>\n",
       "      <td>552.925562</td>\n",
       "      <td>531.805714</td>\n",
       "      <td>561.520848</td>\n",
       "      <td>646.830501</td>\n",
       "      <td>546.531955</td>\n",
       "      <td>511.358494</td>\n",
       "      <td>618.642493</td>\n",
       "      <td>466.028120</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AT1G03090</td>\n",
       "      <td>56.489909</td>\n",
       "      <td>413.232207</td>\n",
       "      <td>115.251591</td>\n",
       "      <td>256.182647</td>\n",
       "      <td>92.072337</td>\n",
       "      <td>138.823846</td>\n",
       "      <td>76.020536</td>\n",
       "      <td>211.405346</td>\n",
       "      <td>51.508880</td>\n",
       "      <td>178.786630</td>\n",
       "      <td>43.954197</td>\n",
       "      <td>89.647417</td>\n",
       "      <td>55.002692</td>\n",
       "      <td>MCCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AT1G04610</td>\n",
       "      <td>555.842721</td>\n",
       "      <td>334.214428</td>\n",
       "      <td>515.020756</td>\n",
       "      <td>389.253013</td>\n",
       "      <td>504.686718</td>\n",
       "      <td>593.003060</td>\n",
       "      <td>737.095889</td>\n",
       "      <td>515.249415</td>\n",
       "      <td>602.295081</td>\n",
       "      <td>566.374468</td>\n",
       "      <td>633.293184</td>\n",
       "      <td>449.895363</td>\n",
       "      <td>435.154338</td>\n",
       "      <td>YUC3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AT1G04990</td>\n",
       "      <td>442.063076</td>\n",
       "      <td>305.687575</td>\n",
       "      <td>386.743703</td>\n",
       "      <td>306.571326</td>\n",
       "      <td>360.500173</td>\n",
       "      <td>332.170769</td>\n",
       "      <td>373.965190</td>\n",
       "      <td>420.078045</td>\n",
       "      <td>374.790222</td>\n",
       "      <td>349.300187</td>\n",
       "      <td>377.968463</td>\n",
       "      <td>422.188547</td>\n",
       "      <td>398.544092</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AT1G05560</td>\n",
       "      <td>820.899447</td>\n",
       "      <td>380.128909</td>\n",
       "      <td>695.339515</td>\n",
       "      <td>406.596100</td>\n",
       "      <td>845.046569</td>\n",
       "      <td>528.074246</td>\n",
       "      <td>632.583643</td>\n",
       "      <td>654.978333</td>\n",
       "      <td>966.961101</td>\n",
       "      <td>577.220749</td>\n",
       "      <td>965.881525</td>\n",
       "      <td>917.404750</td>\n",
       "      <td>805.241328</td>\n",
       "      <td>UGT1, UGT75B1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      GeneID     t0(Fe+)    t6 (Fe+)    t6 (Fe-)   t12 (Fe+)   t12 (Fe-)  \\\n",
       "0  AT1G03070  671.339306  346.370569  534.558117  544.918031  548.079335   \n",
       "1  AT1G03090   56.489909  413.232207  115.251591  256.182647   92.072337   \n",
       "2  AT1G04610  555.842721  334.214428  515.020756  389.253013  504.686718   \n",
       "3  AT1G04990  442.063076  305.687575  386.743703  306.571326  360.500173   \n",
       "4  AT1G05560  820.899447  380.128909  695.339515  406.596100  845.046569   \n",
       "\n",
       "    t18 (Fe+)   t18 (Fe-)   t24 (Fe+)   t24 (Fe-)   t30 (Fe+)   t30 (Fe-)  \\\n",
       "0  552.925562  531.805714  561.520848  646.830501  546.531955  511.358494   \n",
       "1  138.823846   76.020536  211.405346   51.508880  178.786630   43.954197   \n",
       "2  593.003060  737.095889  515.249415  602.295081  566.374468  633.293184   \n",
       "3  332.170769  373.965190  420.078045  374.790222  349.300187  377.968463   \n",
       "4  528.074246  632.583643  654.978333  966.961101  577.220749  965.881525   \n",
       "\n",
       "    t36 (Fe+)    t36(Fe-)       GeneName  \n",
       "0  618.642493  466.028120        unknown  \n",
       "1   89.647417   55.002692           MCCA  \n",
       "2  449.895363  435.154338           YUC3  \n",
       "3  422.188547  398.544092        unknown  \n",
       "4  917.404750  805.241328  UGT1, UGT75B1  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DE_genes_epidermis_RPKM.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(806, 15)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DE_genes_epidermis_RPKM.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/home/temccrac/Programs/git_clones/ECE759_Project/cellular_clarity/SRARunTable.csv'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[8], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m SRA_metadata \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mread_csv(os\u001b[38;5;241m.\u001b[39mpath\u001b[38;5;241m.\u001b[39mjoin(data_path, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mSRARunTable.csv\u001b[39m\u001b[38;5;124m'\u001b[39m))\n\u001b[1;32m      2\u001b[0m SRA_metadata\u001b[38;5;241m.\u001b[39mcolumns\n",
      "File \u001b[0;32m~/Downloads/anaconda3/envs/transcriptomics/lib/python3.12/site-packages/pandas/io/parsers/readers.py:1026\u001b[0m, in \u001b[0;36mread_csv\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, date_format, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options, dtype_backend)\u001b[0m\n\u001b[1;32m   1013\u001b[0m kwds_defaults \u001b[38;5;241m=\u001b[39m _refine_defaults_read(\n\u001b[1;32m   1014\u001b[0m     dialect,\n\u001b[1;32m   1015\u001b[0m     delimiter,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   1022\u001b[0m     dtype_backend\u001b[38;5;241m=\u001b[39mdtype_backend,\n\u001b[1;32m   1023\u001b[0m )\n\u001b[1;32m   1024\u001b[0m kwds\u001b[38;5;241m.\u001b[39mupdate(kwds_defaults)\n\u001b[0;32m-> 1026\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m _read(filepath_or_buffer, kwds)\n",
      "File \u001b[0;32m~/Downloads/anaconda3/envs/transcriptomics/lib/python3.12/site-packages/pandas/io/parsers/readers.py:620\u001b[0m, in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    617\u001b[0m _validate_names(kwds\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mnames\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m))\n\u001b[1;32m    619\u001b[0m \u001b[38;5;66;03m# Create the parser.\u001b[39;00m\n\u001b[0;32m--> 620\u001b[0m parser \u001b[38;5;241m=\u001b[39m TextFileReader(filepath_or_buffer, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwds)\n\u001b[1;32m    622\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m chunksize \u001b[38;5;129;01mor\u001b[39;00m iterator:\n\u001b[1;32m    623\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m parser\n",
      "File \u001b[0;32m~/Downloads/anaconda3/envs/transcriptomics/lib/python3.12/site-packages/pandas/io/parsers/readers.py:1620\u001b[0m, in \u001b[0;36mTextFileReader.__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m   1617\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moptions[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m kwds[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[1;32m   1619\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles: IOHandles \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m-> 1620\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_make_engine(f, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mengine)\n",
      "File \u001b[0;32m~/Downloads/anaconda3/envs/transcriptomics/lib/python3.12/site-packages/pandas/io/parsers/readers.py:1880\u001b[0m, in \u001b[0;36mTextFileReader._make_engine\u001b[0;34m(self, f, engine)\u001b[0m\n\u001b[1;32m   1878\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m mode:\n\u001b[1;32m   1879\u001b[0m         mode \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m-> 1880\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;241m=\u001b[39m get_handle(\n\u001b[1;32m   1881\u001b[0m     f,\n\u001b[1;32m   1882\u001b[0m     mode,\n\u001b[1;32m   1883\u001b[0m     encoding\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moptions\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mencoding\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m),\n\u001b[1;32m   1884\u001b[0m     compression\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moptions\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mcompression\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m),\n\u001b[1;32m   1885\u001b[0m     memory_map\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moptions\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmemory_map\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mFalse\u001b[39;00m),\n\u001b[1;32m   1886\u001b[0m     is_text\u001b[38;5;241m=\u001b[39mis_text,\n\u001b[1;32m   1887\u001b[0m     errors\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moptions\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mencoding_errors\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mstrict\u001b[39m\u001b[38;5;124m\"\u001b[39m),\n\u001b[1;32m   1888\u001b[0m     storage_options\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moptions\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mstorage_options\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m),\n\u001b[1;32m   1889\u001b[0m )\n\u001b[1;32m   1890\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m   1891\u001b[0m f \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles\u001b[38;5;241m.\u001b[39mhandle\n",
      "File \u001b[0;32m~/Downloads/anaconda3/envs/transcriptomics/lib/python3.12/site-packages/pandas/io/common.py:873\u001b[0m, in \u001b[0;36mget_handle\u001b[0;34m(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)\u001b[0m\n\u001b[1;32m    868\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(handle, \u001b[38;5;28mstr\u001b[39m):\n\u001b[1;32m    869\u001b[0m     \u001b[38;5;66;03m# Check whether the filename is to be opened in binary mode.\u001b[39;00m\n\u001b[1;32m    870\u001b[0m     \u001b[38;5;66;03m# Binary mode does not support 'encoding' and 'newline'.\u001b[39;00m\n\u001b[1;32m    871\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mencoding \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mmode:\n\u001b[1;32m    872\u001b[0m         \u001b[38;5;66;03m# Encoding\u001b[39;00m\n\u001b[0;32m--> 873\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mopen\u001b[39m(\n\u001b[1;32m    874\u001b[0m             handle,\n\u001b[1;32m    875\u001b[0m             ioargs\u001b[38;5;241m.\u001b[39mmode,\n\u001b[1;32m    876\u001b[0m             encoding\u001b[38;5;241m=\u001b[39mioargs\u001b[38;5;241m.\u001b[39mencoding,\n\u001b[1;32m    877\u001b[0m             errors\u001b[38;5;241m=\u001b[39merrors,\n\u001b[1;32m    878\u001b[0m             newline\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m    879\u001b[0m         )\n\u001b[1;32m    880\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    881\u001b[0m         \u001b[38;5;66;03m# Binary mode\u001b[39;00m\n\u001b[1;32m    882\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mopen\u001b[39m(handle, ioargs\u001b[38;5;241m.\u001b[39mmode)\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/home/temccrac/Programs/git_clones/ECE759_Project/cellular_clarity/SRARunTable.csv'"
     ]
    }
   ],
   "source": [
    "SRA_metadata = pd.read_csv(os.path.join(git_data_path, 'SRARunTable.csv'))\n",
    "SRA_metadata.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Testing RPKM Properties"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "example_data_path = os.path.join(data_path, 'example_data')\n",
    "X_id = pd.read_csv(os.path.join(example_data_path, 'X_id.csv'))\n",
    "Y = pd.read_csv(os.path.join(example_data_path, 'Y.csv.gz'), compression='gzip', index_col = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Simulate RPKM \n",
    "import numpy as np\n",
    "num_genes = len(Y.columns) # number of genes, columns are genes and rows are samples\n",
    "gene_lengths = np.random.randint(500, 5000, size=num_genes) # get random gene lengths\n",
    "total_mapped_reads = Y.sum(axis=0) # total amount of each gene detected\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Processing Raw RNA-Seq Data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The folders in arabidopsis_data should all contain fastq.gz files for adapter trimming"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Directories: ['/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long']\n",
      "Files: ['/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/GSL_100B_Long_PoolD-001.tar', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Sample - Pool Information_1-19.xlsx', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/GSL_100B_Long_PoolA-002.tar', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/GSL_100B_Long_PoolB-003.tar', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/GSL_100B_Long_PoolC-005.tar', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/GSL_105B_Long-004.tar', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_18_S104_L004_R1_001_fastqc.zip', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_11_S109_L004_R1_001_fastqc.html', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_4_S102_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_11_S109_L004_R1_001_fastqc.zip', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_3_S101_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_4_S102_L004_R1_001_fastqc.zip', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_2_S100_L004_R1_001_fastqc.zip', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_10_S108_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_2_S100_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_3_S101_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_12_S110_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_10_S108_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_11_S109_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_18_S104_L004_R1_001_fastqc.html', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_5_S103_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_18_S104_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_7_S105_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_8_S106_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_8_S106_L004_R1_001_fastqc.html', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_1_S99_L004_R1_001_fastqc.zip', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_9_S107_L004_R1_001_fastqc.html', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_5_S103_L004_R1_001_fastqc.html', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_1_S99_L004_R1_001_fastqc.html', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_7_S105_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_10_S108_L004_R1_001_fastqc.html', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_18_S104_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_7_S105_L004_R1_001_fastqc.html', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_8_S106_L004_R1_001_fastqc.zip', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_2_S100_L004_R1_001_fastqc.html', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_2_S100_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_12_S110_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_4_S102_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_11_S109_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_4_S102_L004_R1_001_fastqc.html', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_9_S107_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_9_S107_L004_R1_001_fastqc.zip', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_3_S101_L004_R1_001_fastqc.zip', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_10_S108_L004_R1_001_fastqc.zip', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_8_S106_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_1_S99_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_7_S105_L004_R1_001_fastqc.zip', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_5_S103_L004_R1_001_fastqc.zip', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_9_S107_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_3_S101_L004_R1_001_fastqc.html', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_1_S99_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_12_S110_L004_R1_001_fastqc.html', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_5_S103_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_12_S110_L004_R1_001_fastqc.zip', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_28_S114_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_19_S117_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_27_S113_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_13_S111_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_22_S119_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_21_S118_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_28_S114_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_22_S119_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_26_S112_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_6_S116_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_21_S118_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_6_S116_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_13_S111_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_19_S117_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_27_S113_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_29_S115_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_29_S115_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_26_S112_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_indx12_S123_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_33_S120_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_37_S125_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_indx12_S123_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_32_S127_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_31_S126_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_34_S121_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_indx11_S122_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_30_S124_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_34_S121_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_30_S124_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_33_S120_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_32_S127_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_indx11_S122_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_31_S126_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_37_S125_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_25_S128_L007_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_20_S133_L007_R1_001_fastqc.html', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_16_S131_L007_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_16_S131_L007_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_20_S133_L007_R1_001_fastqc.html', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_25_S128_L007_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_indx2_S129_L007_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_17_S132_L007_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_20_S133_L007_R1_001_fastqc.zip', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_indx3_S130_L007_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_indx2_S129_L007_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_indx3_S130_L007_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_20_S133_L007_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_20_S133_L007_R1_001_fastqc.zip', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_17_S132_L007_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_20_S133_L007_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_23_S5_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_38_S7_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_35_S9_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/38_S7_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_14_S3_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_24_S6_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/24_S6_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_36_S10_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/39_S8_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/36_S10_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/14_S3_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_15_S4_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/15_S4_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_39_S8_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/23_S5_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/35_S9_L005_R1_001.fastq.gz']\n"
     ]
    }
   ],
   "source": [
    "# Gather the file paths and directories \n",
    "sequence_path = os.path.join(data_path, \"adrabidopsis_data\")\n",
    "directories = []\n",
    "file_paths = []\n",
    "dir_names = []\n",
    "\n",
    "for root, dirs, files in os.walk(sequence_path):\n",
    "    # Append full path of each directory\n",
    "    for dir_name in dirs:\n",
    "        dir_names.append(dir_name)\n",
    "        directories.append(os.path.join(root, dir_name))\n",
    "    \n",
    "    # Append full path of each file\n",
    "    for file in files:\n",
    "        file_paths.append(os.path.join(root, file))\n",
    "\n",
    "print(\"Directories:\", directories)\n",
    "print(\"Files:\", file_paths)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Lane4_FastqFiles': ['/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_4_S102_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_3_S101_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_10_S108_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_2_S100_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_3_S101_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_12_S110_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_10_S108_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_11_S109_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_5_S103_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_18_S104_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_7_S105_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_8_S106_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_7_S105_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_18_S104_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_2_S100_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_12_S110_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_4_S102_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_11_S109_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_9_S107_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_8_S106_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_1_S99_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_9_S107_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_1_S99_L004_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_5_S103_L004_R1_001.fastq.gz'],\n",
       " 'Lane5_FastqFiles': ['/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_28_S114_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_19_S117_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_27_S113_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_13_S111_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_22_S119_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_21_S118_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_28_S114_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_22_S119_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_26_S112_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_6_S116_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_21_S118_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_6_S116_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_13_S111_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_19_S117_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_27_S113_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_29_S115_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_29_S115_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_26_S112_L005_R1_001.fastq.gz'],\n",
       " 'Lane6_FastqFiles': ['/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_indx12_S123_L006_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_33_S120_L006_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_37_S125_L006_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_indx12_S123_L006_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_32_S127_L006_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_31_S126_L006_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_34_S121_L006_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_indx11_S122_L006_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_30_S124_L006_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_34_S121_L006_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_30_S124_L006_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_33_S120_L006_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_32_S127_L006_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_indx11_S122_L006_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_31_S126_L006_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_37_S125_L006_R1_001.fastq.gz'],\n",
       " 'Lane7_FastqFiles': ['/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_25_S128_L007_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_16_S131_L007_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_16_S131_L007_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_25_S128_L007_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_indx2_S129_L007_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_17_S132_L007_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_indx3_S130_L007_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_indx2_S129_L007_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_indx3_S130_L007_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_20_S133_L007_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_17_S132_L007_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_20_S133_L007_R1_001.fastq.gz'],\n",
       " '20171212_GSL_105B_Long': ['/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_23_S5_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_38_S7_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_35_S9_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/38_S7_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_14_S3_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_24_S6_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/24_S6_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_36_S10_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/39_S8_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/36_S10_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/14_S3_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_15_S4_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/15_S4_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_39_S8_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/23_S5_L005_R1_001.fastq.gz',\n",
       "  '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/35_S9_L005_R1_001.fastq.gz']}"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Store each fastq.gz file containing sequences under its respective sequencing pool directory\n",
    "sequence_dictionary = dict.fromkeys(dir_names)\n",
    "raw_sequence_file_paths = []\n",
    "for n, dir in enumerate(directories): \n",
    "    raw_sequence_file_paths = []  # Reset for each directory\n",
    "    for root, dirs, files in os.walk(dir):\n",
    "        for file in files: \n",
    "            if file.endswith('fastq.gz'): \n",
    "                raw_sequence_file_paths.append(os.path.join(dir,file))\n",
    "    sequence_dictionary[dir_names[n]] = raw_sequence_file_paths\n",
    "sequence_dictionary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_4_S102_L004_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_4_S102_L004_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 115505 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_4_S102_L004_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 15736335\n",
      "Too short after clip: 619\n",
      "Clipped 'end' reads: Count: 10316007, Mean: 28.73, Sd: 26.86\n",
      "Trimmed 1008061 reads by an average of 2.88 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_4_S102_L004_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_4_S102_L004_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_2_S100_L004_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_2_S100_L004_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 99960 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_2_S100_L004_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 11099096\n",
      "Too short after clip: 522\n",
      "Clipped 'end' reads: Count: 6837162, Mean: 26.88, Sd: 27.12\n",
      "Trimmed 742689 reads by an average of 2.81 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_2_S100_L004_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_2_S100_L004_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_3_S101_L004_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_3_S101_L004_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 113324 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_3_S101_L004_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 8287553\n",
      "Too short after clip: 1084\n",
      "Clipped 'end' reads: Count: 5363000, Mean: 30.24, Sd: 28.54\n",
      "Trimmed 602621 reads by an average of 3.09 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_3_S101_L004_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_3_S101_L004_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_10_S108_L004_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_10_S108_L004_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 105414 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_10_S108_L004_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 10467574\n",
      "Too short after clip: 390\n",
      "Clipped 'end' reads: Count: 6631659, Mean: 27.25, Sd: 26.58\n",
      "Trimmed 652156 reads by an average of 2.88 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_10_S108_L004_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_10_S108_L004_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_11_S109_L004_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_11_S109_L004_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 110615 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_11_S109_L004_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 9518766\n",
      "Too short after clip: 1887\n",
      "Clipped 'end' reads: Count: 6149396, Mean: 27.74, Sd: 26.56\n",
      "Trimmed 623177 reads by an average of 2.85 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_11_S109_L004_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_11_S109_L004_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_5_S103_L004_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_5_S103_L004_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 99769 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_5_S103_L004_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 17687375\n",
      "Too short after clip: 625\n",
      "Clipped 'end' reads: Count: 10889525, Mean: 26.49, Sd: 26.82\n",
      "Trimmed 1109212 reads by an average of 2.88 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_5_S103_L004_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_5_S103_L004_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_18_S104_L004_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_18_S104_L004_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 114042 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_18_S104_L004_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 26800719\n",
      "Too short after clip: 1092\n",
      "Clipped 'end' reads: Count: 17554098, Mean: 28.35, Sd: 26.58\n",
      "Trimmed 1700759 reads by an average of 2.85 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_18_S104_L004_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_18_S104_L004_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_7_S105_L004_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_7_S105_L004_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 116763 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_7_S105_L004_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 20704178\n",
      "Too short after clip: 917\n",
      "Clipped 'end' reads: Count: 13584602, Mean: 29.44, Sd: 27.35\n",
      "Trimmed 1354754 reads by an average of 2.96 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_7_S105_L004_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_7_S105_L004_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_12_S110_L004_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_12_S110_L004_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 110381 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_12_S110_L004_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 25740212\n",
      "Too short after clip: 2159\n",
      "Clipped 'end' reads: Count: 16602024, Mean: 27.77, Sd: 26.51\n",
      "Trimmed 1633383 reads by an average of 2.91 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_12_S110_L004_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_12_S110_L004_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_8_S106_L004_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_8_S106_L004_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 134240 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_8_S106_L004_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 12902327\n",
      "Too short after clip: 996\n",
      "Clipped 'end' reads: Count: 8964755, Mean: 32.83, Sd: 28.12\n",
      "Trimmed 952111 reads by an average of 3.12 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_8_S106_L004_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_8_S106_L004_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_1_S99_L004_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_1_S99_L004_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 118055 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_1_S99_L004_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 21695861\n",
      "Too short after clip: 915\n",
      "Clipped 'end' reads: Count: 14318118, Mean: 29.62, Sd: 27.27\n",
      "Trimmed 1486401 reads by an average of 2.86 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_1_S99_L004_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_1_S99_L004_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_9_S107_L004_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_9_S107_L004_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 106452 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_9_S107_L004_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 11461407\n",
      "Too short after clip: 459\n",
      "Clipped 'end' reads: Count: 7310991, Mean: 27.01, Sd: 26.27\n",
      "Trimmed 702235 reads by an average of 2.84 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/Long_9_S107_L004_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_9_S107_L004_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_27_S113_L005_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_27_S113_L005_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 114138 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_27_S113_L005_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 25565801\n",
      "Too short after clip: 812\n",
      "Clipped 'end' reads: Count: 16690784, Mean: 29.38, Sd: 27.36\n",
      "Trimmed 2029301 reads by an average of 2.48 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_27_S113_L005_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_27_S113_L005_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_22_S119_L005_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_22_S119_L005_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 120969 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_22_S119_L005_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 28599265\n",
      "Too short after clip: 779\n",
      "Clipped 'end' reads: Count: 19135731, Mean: 30.45, Sd: 27.43\n",
      "Trimmed 2337556 reads by an average of 2.55 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_22_S119_L005_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_22_S119_L005_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_28_S114_L005_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_28_S114_L005_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 112570 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_28_S114_L005_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 28050297\n",
      "Too short after clip: 803\n",
      "Clipped 'end' reads: Count: 18265395, Mean: 28.41, Sd: 26.72\n",
      "Trimmed 2138383 reads by an average of 2.42 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_28_S114_L005_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_28_S114_L005_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_6_S116_L005_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_6_S116_L005_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 121750 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_6_S116_L005_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 27603146\n",
      "Too short after clip: 1443\n",
      "Clipped 'end' reads: Count: 18623318, Mean: 29.99, Sd: 26.94\n",
      "Trimmed 2192242 reads by an average of 2.47 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_6_S116_L005_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_6_S116_L005_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_21_S118_L005_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_21_S118_L005_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 106905 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_21_S118_L005_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 38132556\n",
      "Too short after clip: 1543\n",
      "Clipped 'end' reads: Count: 24241329, Mean: 28.70, Sd: 27.73\n",
      "Trimmed 3123654 reads by an average of 2.50 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_21_S118_L005_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_21_S118_L005_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_13_S111_L005_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_13_S111_L005_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 101958 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_13_S111_L005_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 27048756\n",
      "Too short after clip: 762\n",
      "Clipped 'end' reads: Count: 17014031, Mean: 25.96, Sd: 25.75\n",
      "Trimmed 1952519 reads by an average of 2.26 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_13_S111_L005_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_13_S111_L005_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_19_S117_L005_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_19_S117_L005_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 121618 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_19_S117_L005_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 29485732\n",
      "Too short after clip: 1162\n",
      "Clipped 'end' reads: Count: 19891844, Mean: 30.03, Sd: 27.16\n",
      "Trimmed 2273477 reads by an average of 2.56 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_19_S117_L005_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_19_S117_L005_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_29_S115_L005_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_29_S115_L005_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 103383 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_29_S115_L005_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 32662714\n",
      "Too short after clip: 665\n",
      "Clipped 'end' reads: Count: 20573486, Mean: 26.61, Sd: 26.25\n",
      "Trimmed 2362723 reads by an average of 2.35 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_29_S115_L005_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_29_S115_L005_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_26_S112_L005_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_26_S112_L005_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 106321 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_26_S112_L005_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 7219166\n",
      "Too short after clip: 200\n",
      "Clipped 'end' reads: Count: 4580365, Mean: 27.98, Sd: 27.06\n",
      "Trimmed 586210 reads by an average of 2.41 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/Long_26_S112_L005_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_26_S112_L005_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_indx12_S123_L006_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_indx12_S123_L006_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 121782 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_indx12_S123_L006_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 41837016\n",
      "Too short after clip: 2013\n",
      "Clipped 'end' reads: Count: 27885599, Mean: 30.33, Sd: 27.29\n",
      "Trimmed 3067326 reads by an average of 2.82 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_indx12_S123_L006_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_indx12_S123_L006_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_32_S127_L006_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_32_S127_L006_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 107276 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_32_S127_L006_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 32456078\n",
      "Too short after clip: 827\n",
      "Clipped 'end' reads: Count: 20765251, Mean: 26.77, Sd: 25.89\n",
      "Trimmed 2026136 reads by an average of 2.67 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_32_S127_L006_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_32_S127_L006_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_34_S121_L006_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_34_S121_L006_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 105352 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_34_S121_L006_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 15410771\n",
      "Too short after clip: 452\n",
      "Clipped 'end' reads: Count: 9768696, Mean: 26.37, Sd: 25.77\n",
      "Trimmed 1011357 reads by an average of 2.66 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_34_S121_L006_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_34_S121_L006_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_30_S124_L006_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_30_S124_L006_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 118660 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_30_S124_L006_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 15634357\n",
      "Too short after clip: 476\n",
      "Clipped 'end' reads: Count: 10351675, Mean: 29.09, Sd: 26.78\n",
      "Trimmed 1078758 reads by an average of 2.65 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_30_S124_L006_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_30_S124_L006_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_33_S120_L006_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_33_S120_L006_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 101272 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_33_S120_L006_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 43040130\n",
      "Too short after clip: 1522\n",
      "Clipped 'end' reads: Count: 26831453, Mean: 26.01, Sd: 25.97\n",
      "Trimmed 2824372 reads by an average of 2.69 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_33_S120_L006_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_33_S120_L006_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_indx11_S122_L006_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_indx11_S122_L006_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 98162 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_indx11_S122_L006_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 35709585\n",
      "Too short after clip: 1151\n",
      "Clipped 'end' reads: Count: 21962931, Mean: 26.04, Sd: 26.40\n",
      "Trimmed 2431142 reads by an average of 2.60 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_indx11_S122_L006_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_indx11_S122_L006_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_31_S126_L006_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_31_S126_L006_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 106071 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_31_S126_L006_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 18621549\n",
      "Too short after clip: 596\n",
      "Clipped 'end' reads: Count: 11796349, Mean: 27.02, Sd: 26.42\n",
      "Trimmed 1177384 reads by an average of 2.66 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_31_S126_L006_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_31_S126_L006_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_37_S125_L006_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_37_S125_L006_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 106753 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_37_S125_L006_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 17989828\n",
      "Too short after clip: 1249\n",
      "Clipped 'end' reads: Count: 11451395, Mean: 26.67, Sd: 26.00\n",
      "Trimmed 1184189 reads by an average of 2.65 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/Long_37_S125_L006_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_37_S125_L006_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_25_S128_L007_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_25_S128_L007_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 123329 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_25_S128_L007_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 32103337\n",
      "Too short after clip: 1085\n",
      "Clipped 'end' reads: Count: 21549292, Mean: 30.20, Sd: 27.20\n",
      "Trimmed 2313356 reads by an average of 2.58 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_25_S128_L007_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_25_S128_L007_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_16_S131_L007_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_16_S131_L007_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 106459 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_16_S131_L007_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 23458663\n",
      "Too short after clip: 382\n",
      "Clipped 'end' reads: Count: 14984291, Mean: 25.95, Sd: 25.29\n",
      "Trimmed 1381818 reads by an average of 2.40 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_16_S131_L007_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_16_S131_L007_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_indx2_S129_L007_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_indx2_S129_L007_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 107412 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_indx2_S129_L007_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 65823383\n",
      "Too short after clip: 3407\n",
      "Clipped 'end' reads: Count: 41962454, Mean: 26.73, Sd: 25.93\n",
      "Trimmed 4372297 reads by an average of 2.53 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_indx2_S129_L007_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_indx2_S129_L007_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_17_S132_L007_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_17_S132_L007_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 110995 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_17_S132_L007_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 23638613\n",
      "Too short after clip: 1543\n",
      "Clipped 'end' reads: Count: 15276311, Mean: 27.10, Sd: 25.86\n",
      "Trimmed 1487153 reads by an average of 2.62 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_17_S132_L007_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_17_S132_L007_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_indx3_S130_L007_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_indx3_S130_L007_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 100103 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_indx3_S130_L007_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 56963664\n",
      "Too short after clip: 1854\n",
      "Clipped 'end' reads: Count: 35262526, Mean: 25.86, Sd: 26.02\n",
      "Trimmed 3528279 reads by an average of 2.55 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_indx3_S130_L007_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_indx3_S130_L007_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_20_S133_L007_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_20_S133_L007_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 100887 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_20_S133_L007_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 33049059\n",
      "Too short after clip: 864\n",
      "Clipped 'end' reads: Count: 20453763, Mean: 26.46, Sd: 26.54\n",
      "Trimmed 2037528 reads by an average of 2.56 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/Long_20_S133_L007_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_20_S133_L007_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/38_S7_L005_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_38_S7_L005_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 92396 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/38_S7_L005_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 14840437\n",
      "Too short after clip: 694\n",
      "Clipped 'end' reads: Count: 9088891, Mean: 23.08, Sd: 23.88\n",
      "Trimmed 916763 reads by an average of 2.35 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/38_S7_L005_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_38_S7_L005_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/24_S6_L005_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_24_S6_L005_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 104136 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/24_S6_L005_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 28953834\n",
      "Too short after clip: 563\n",
      "Clipped 'end' reads: Count: 18541192, Mean: 25.09, Sd: 24.46\n",
      "Trimmed 1701867 reads by an average of 2.42 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/24_S6_L005_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_24_S6_L005_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/39_S8_L005_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_39_S8_L005_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 80566 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/39_S8_L005_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 114001497\n",
      "Too short after clip: 4303\n",
      "Clipped 'end' reads: Count: 66125587, Mean: 21.61, Sd: 23.94\n",
      "Trimmed 6461807 reads by an average of 2.33 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/39_S8_L005_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_39_S8_L005_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/36_S10_L005_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_36_S10_L005_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 93195 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/36_S10_L005_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 18811726\n",
      "Too short after clip: 675\n",
      "Clipped 'end' reads: Count: 11429738, Mean: 24.66, Sd: 25.46\n",
      "Trimmed 1216552 reads by an average of 2.48 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/36_S10_L005_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_36_S10_L005_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/14_S3_L005_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_14_S3_L005_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 102041 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/14_S3_L005_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 7519570\n",
      "Too short after clip: 216\n",
      "Clipped 'end' reads: Count: 4727577, Mean: 26.19, Sd: 25.98\n",
      "Trimmed 509950 reads by an average of 2.42 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/14_S3_L005_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_14_S3_L005_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/15_S4_L005_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_15_S4_L005_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 111770 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/15_S4_L005_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 7875354\n",
      "Too short after clip: 314\n",
      "Clipped 'end' reads: Count: 5133772, Mean: 28.09, Sd: 26.52\n",
      "Trimmed 537187 reads by an average of 2.53 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/15_S4_L005_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_15_S4_L005_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/23_S5_L005_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_23_S5_L005_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 82605 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/23_S5_L005_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 20553749\n",
      "Too short after clip: 486\n",
      "Clipped 'end' reads: Count: 11985262, Mean: 22.39, Sd: 24.45\n",
      "Trimmed 1232151 reads by an average of 2.30 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/23_S5_L005_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_23_S5_L005_R1_001.fastq.gz\n",
      "Command Line: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/35_S9_L005_R1_001.fastq.gz -o /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_35_S9_L005_R1_001.fastq.gz -l 20 -q 15 -x 0.75\n",
      "Scale used: 2.2\n",
      "Phred: 33\n",
      "Threshold used: 751 out of 300000\n",
      "Adapter Illumina_Universal_Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC): counted 104562 at the 'end' of '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/35_S9_L005_R1_001.fastq.gz', clip set to 1\n",
      "Files: 1\n",
      "Total reads: 17998644\n",
      "Too short after clip: 721\n",
      "Clipped 'end' reads: Count: 11450292, Mean: 26.04, Sd: 25.50\n",
      "Trimmed 1260684 reads by an average of 2.44 bases on quality < 15\n",
      "Trimmed: /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/35_S9_L005_R1_001.fastq.gz → /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_35_S9_L005_R1_001.fastq.gz\n",
      "\n",
      "Trimmed Sequences Dictionary:\n",
      "Lane4_FastqFiles: ['/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_4_S102_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_2_S100_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_3_S101_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_10_S108_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_11_S109_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_5_S103_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_18_S104_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_7_S105_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_12_S110_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_8_S106_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_1_S99_L004_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_9_S107_L004_R1_001.fastq.gz']\n",
      "Lane5_FastqFiles: ['/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_27_S113_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_22_S119_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_28_S114_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_6_S116_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_21_S118_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_13_S111_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_19_S117_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_29_S115_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_26_S112_L005_R1_001.fastq.gz']\n",
      "Lane6_FastqFiles: ['/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_indx12_S123_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_32_S127_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_34_S121_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_30_S124_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_33_S120_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_indx11_S122_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_31_S126_L006_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_37_S125_L006_R1_001.fastq.gz']\n",
      "Lane7_FastqFiles: ['/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_25_S128_L007_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_16_S131_L007_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_indx2_S129_L007_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_17_S132_L007_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_indx3_S130_L007_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_20_S133_L007_R1_001.fastq.gz']\n",
      "20171212_GSL_105B_Long: ['/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_38_S7_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_24_S6_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_39_S8_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_36_S10_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_14_S3_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_15_S4_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_23_S5_L005_R1_001.fastq.gz', '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_35_S9_L005_R1_001.fastq.gz']\n"
     ]
    }
   ],
   "source": [
    "# Trim all of the files using the fastq-mcf command\n",
    "import subprocess\n",
    "\n",
    "# Adapter file path\n",
    "adapter_path = \"/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/adapters.fa\"\n",
    "\n",
    "# Dictionary to store trimmed file paths\n",
    "trimmed_files = {}\n",
    "\n",
    "# Process each directory separately\n",
    "for pool, file_paths in sequence_dictionary.items():\n",
    "    trimmed_files[pool] = []  # Store trimmed file paths for this pool\n",
    "\n",
    "    for file_path in file_paths:\n",
    "        # Get directory and filename\n",
    "        directory, filename = os.path.split(file_path)\n",
    "\n",
    "        # Generate new filename with \"trimmed_\" prefix\n",
    "        trimmed_filename = f\"trimmed_{filename}\"\n",
    "        trimmed_path = os.path.join(directory, trimmed_filename)\n",
    "\n",
    "        # Construct the fastq-mcf command\n",
    "        command = [\n",
    "            \"fastq-mcf\",\n",
    "            adapter_path,\n",
    "            file_path,\n",
    "            \"-o\", trimmed_path,\n",
    "            \"-l\", \"20\", \"-q\", \"15\", \"-x\", \"0.75\"\n",
    "        ]\n",
    "\n",
    "        # Run the trimming process\n",
    "        subprocess.run(command, check=True)\n",
    "        print(f\"Trimmed: {file_path} → {trimmed_path}\")\n",
    "\n",
    "        # Store the trimmed file path\n",
    "        trimmed_files[pool].append(trimmed_path)\n",
    "\n",
    "# Print summary of trimmed files\n",
    "print(\"\\nTrimmed Sequences Dictionary:\")\n",
    "for pool, files in trimmed_files.items():\n",
    "    print(f\"{pool}: {files}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_3_S101_L004_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_10_S108_L004_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_12_S110_L004_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_8_S106_L004_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_7_S105_L004_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_18_S104_L004_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_2_S100_L004_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_4_S102_L004_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_11_S109_L004_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_9_S107_L004_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_1_S99_L004_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_5_S103_L004_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_28_S114_L005_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_19_S117_L005_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_13_S111_L005_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_21_S118_L005_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_22_S119_L005_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_26_S112_L005_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_6_S116_L005_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_27_S113_L005_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_29_S115_L005_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_33_S120_L006_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_37_S125_L006_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_indx12_S123_L006_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_31_S126_L006_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_34_S121_L006_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_indx11_S122_L006_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_30_S124_L006_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane6_FastqFiles/trimmed_Long_32_S127_L006_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_16_S131_L007_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_25_S128_L007_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_indx2_S129_L007_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_indx3_S130_L007_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_17_S132_L007_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane7_FastqFiles/trimmed_Long_20_S133_L007_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_23_S5_L005_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_38_S7_L005_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_35_S9_L005_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_14_S3_L005_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_24_S6_L005_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_36_S10_L005_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_15_S4_L005_R1_001.fastq.gz',\n",
       " '/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/20171212_GSL_105B_Long/trimmed_39_S8_L005_R1_001.fastq.gz']"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Initialize dictionary to store trimmed files by directory\n",
    "trimmed_sequence_dictionary = dict.fromkeys(dir_names)\n",
    "\n",
    "for n, dir in enumerate(directories): \n",
    "    trimmed_file_paths = []  # Reset list for each directory\n",
    "    for root, dirs, files in os.walk(dir):\n",
    "        for file in files: \n",
    "            if \"trimmed_\" in file and file.endswith('fastq.gz'): \n",
    "                trimmed_file_paths.append(os.path.join(dir, file))\n",
    "    \n",
    "    trimmed_sequence_dictionary[dir_names[n]] = trimmed_file_paths\n",
    "\n",
    "# Output the dictionary\n",
    "trimmed_sequence_dictionary\n",
    "fastq_files = [file for files_list in trimmed_sequence_dictionary.values() for file in files_list]\n",
    "fastq_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "trimmed_Long_7_S105_L004_R1_001trimmed_Long_11_S109_L004_R1_001trimmed_Long_1_S99_L004_R1_001trimmed_Long_12_S110_L004_R1_001trimmed_Long_2_S100_L004_R1_001trimmed_Long_28_S114_L005_R1_001trimmed_Long_13_S111_L005_R1_001       now being alignednow being alignednow being alignednow being alignednow being alignednow being aligned\n",
      "now being aligned\n",
      "\n",
      "\n",
      "\n",
      "Running TopHat2 for trimmed_Long_1_S99_L004_R1_001...\n",
      "Running TopHat2 for trimmed_Long_12_S110_L004_R1_001...\n",
      "Running TopHat2 for trimmed_Long_7_S105_L004_R1_001...Running TopHat2 for trimmed_Long_11_S109_L004_R1_001...Running TopHat2 for trimmed_Long_28_S114_L005_R1_001...\n",
      "Running TopHat2 for trimmed_Long_2_S100_L004_R1_001...\n",
      "\n",
      "Running TopHat2 for trimmed_Long_13_S111_L005_R1_001...\n",
      "\n",
      "\n",
      "\n",
      "trimmed_Long_3_S101_L004_R1_001 now being aligned\n",
      "Running TopHat2 for trimmed_Long_3_S101_L004_R1_001...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "[2025-03-15 16:39:51] Beginning TopHat run (v2.1.1)\n",
      "\n",
      "[2025-03-15 16:39:51] Beginning TopHat run (v2.1.1)\n",
      "-----------------------------------------------\n",
      "\n",
      "[2025-03-15 16:39:51] Beginning TopHat run (v2.1.1)\n",
      "[2025-03-15 16:39:51] Checking for Bowtie\n",
      "-----------------------------------------------\n",
      "[2025-03-15 16:39:51] Beginning TopHat run (v2.1.1)\n",
      "-----------------------------------------------\n",
      "[2025-03-15 16:39:51] Checking for Bowtie\n",
      "[2025-03-15 16:39:51] Checking for Bowtie\n",
      "\n",
      "[2025-03-15 16:39:51] Beginning TopHat run (v2.1.1)\n",
      "-----------------------------------------------\n",
      "[2025-03-15 16:39:51] Checking for Bowtie\n",
      "-----------------------------------------------\n",
      "[2025-03-15 16:39:51] Checking for Bowtie\n",
      "\n",
      "[2025-03-15 16:39:51] Beginning TopHat run (v2.1.1)\n",
      "-----------------------------------------------\n",
      "[2025-03-15 16:39:51] Checking for Bowtie\n",
      "\n",
      "[2025-03-15 16:39:51] Beginning TopHat run (v2.1.1)\n",
      "-----------------------------------------------\n",
      "[2025-03-15 16:39:51] Checking for Bowtie\n",
      "\t\t  Bowtie version:\t 2.5.4.0\n",
      "[2025-03-15 16:39:51] Checking for Bowtie index files (genome)..\n",
      "[2025-03-15 16:39:51] Checking for reference FASTA file\n",
      "[2025-03-15 16:39:51] Generating SAM header for /home/temccrac/Programs/data/genomes/TAIR10_index\n",
      "\t\t  Bowtie version:\t 2.5.4.0\n",
      "[2025-03-15 16:39:51] Checking for Bowtie index files (genome)..\n",
      "[2025-03-15 16:39:51] Checking for reference FASTA file\n",
      "[2025-03-15 16:39:51] Generating SAM header for /home/temccrac/Programs/data/genomes/TAIR10_index\n",
      "\t\t  Bowtie version:\t 2.5.4.0\n",
      "[2025-03-15 16:39:51] Checking for Bowtie index files (genome)..\n",
      "[2025-03-15 16:39:51] Checking for reference FASTA file\n",
      "[2025-03-15 16:39:51] Generating SAM header for /home/temccrac/Programs/data/genomes/TAIR10_index\n",
      "\t\t  Bowtie version:\t 2.5.4.0\n",
      "[2025-03-15 16:39:51] Checking for Bowtie index files (genome)..\n",
      "\t\t  Bowtie version:\t 2.5.4.0\n",
      "[2025-03-15 16:39:51] Checking for Bowtie index files (genome)..\n",
      "[2025-03-15 16:39:51] Checking for reference FASTA file\n",
      "[2025-03-15 16:39:51] Generating SAM header for /home/temccrac/Programs/data/genomes/TAIR10_index\n",
      "[2025-03-15 16:39:51] Checking for reference FASTA file\n",
      "[2025-03-15 16:39:51] Generating SAM header for /home/temccrac/Programs/data/genomes/TAIR10_index\n",
      "\t\t  Bowtie version:\t 2.5.4.0\n",
      "[2025-03-15 16:39:51] Checking for Bowtie index files (genome)..\n",
      "[2025-03-15 16:39:51] Checking for reference FASTA file\n",
      "[2025-03-15 16:39:51] Generating SAM header for /home/temccrac/Programs/data/genomes/TAIR10_index\n",
      "\t\t  Bowtie version:\t 2.5.4.0\n",
      "[2025-03-15 16:39:51] Checking for Bowtie index files (genome)..\n",
      "[2025-03-15 16:39:51] Checking for reference FASTA file\n",
      "\n",
      "[2025-03-15 16:39:51] Generating SAM header for /home/temccrac/Programs/data/genomes/TAIR10_index\n",
      "[2025-03-15 16:39:51] Beginning TopHat run (v2.1.1)\n",
      "-----------------------------------------------\n",
      "[2025-03-15 16:39:51] Checking for Bowtie\n",
      "\t\t  Bowtie version:\t 2.5.4.0\n",
      "[2025-03-15 16:39:51] Checking for Bowtie index files (genome)..\n",
      "[2025-03-15 16:39:51] Checking for reference FASTA file\n",
      "[2025-03-15 16:39:51] Generating SAM header for /home/temccrac/Programs/data/genomes/TAIR10_index\n",
      "[2025-03-15 16:39:51] Reading known junctions from GTF file\n",
      "[2025-03-15 16:39:51] Reading known junctions from GTF file\n",
      "[2025-03-15 16:39:51] Reading known junctions from GTF file\n",
      "[2025-03-15 16:39:51] Reading known junctions from GTF file\n",
      "[2025-03-15 16:39:51] Reading known junctions from GTF file\n",
      "[2025-03-15 16:39:51] Reading known junctions from GTF file\n",
      "[2025-03-15 16:39:51] Reading known junctions from GTF file\n",
      "[2025-03-15 16:39:51] Reading known junctions from GTF file\n",
      "[2025-03-15 16:39:54] Preparing reads\n",
      "[2025-03-15 16:39:54] Preparing reads\n",
      "[2025-03-15 16:39:54] Preparing reads\n",
      "[2025-03-15 16:39:54] Preparing reads\n",
      "[2025-03-15 16:39:54] Preparing reads\n",
      "[2025-03-15 16:39:54] Preparing reads\n",
      "[2025-03-15 16:39:54] Preparing reads\n",
      "[2025-03-15 16:39:54] Preparing reads\n",
      "\t left reads: min. length=20, max. length=125, 8285810 kept reads (659 discarded)\n",
      "[2025-03-15 16:41:28] Building transcriptome data files /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_3_S101_L004_R1_001/tmp/Arabidopsis_thaliana.TAIR10.54\n",
      "[2025-03-15 16:41:33] Building Bowtie index from Arabidopsis_thaliana.TAIR10.54.fa\n",
      "\t left reads: min. length=20, max. length=125, 9515102 kept reads (1777 discarded)\n",
      "[2025-03-15 16:41:43] Building transcriptome data files /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_11_S109_L004_R1_001/tmp/Arabidopsis_thaliana.TAIR10.54\n",
      "[2025-03-15 16:41:48] Building Bowtie index from Arabidopsis_thaliana.TAIR10.54.fa\n",
      "\t left reads: min. length=20, max. length=125, 11097547 kept reads (1027 discarded)\n",
      "[2025-03-15 16:42:02] Building transcriptome data files /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_2_S100_L004_R1_001/tmp/Arabidopsis_thaliana.TAIR10.54\n",
      "[2025-03-15 16:42:07] Building Bowtie index from Arabidopsis_thaliana.TAIR10.54.fa\n",
      "\t left reads: min. length=20, max. length=125, 20701728 kept reads (1533 discarded)\n",
      "[2025-03-15 16:43:47] Building transcriptome data files /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_7_S105_L004_R1_001/tmp/Arabidopsis_thaliana.TAIR10.54\n",
      "[2025-03-15 16:43:53] Building Bowtie index from Arabidopsis_thaliana.TAIR10.54.fa\n",
      "\t left reads: min. length=20, max. length=125, 21692826 kept reads (2120 discarded)\n",
      "[2025-03-15 16:44:00] Building transcriptome data files /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_1_S99_L004_R1_001/tmp/Arabidopsis_thaliana.TAIR10.54\n",
      "[2025-03-15 16:44:05] Building Bowtie index from Arabidopsis_thaliana.TAIR10.54.fa\n",
      "\t left reads: min. length=20, max. length=125, 25732997 kept reads (5056 discarded)\n",
      "[2025-03-15 16:44:47] Building transcriptome data files /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_12_S110_L004_R1_001/tmp/Arabidopsis_thaliana.TAIR10.54\n",
      "[2025-03-15 16:44:52] Building Bowtie index from Arabidopsis_thaliana.TAIR10.54.fa\n",
      "\t left reads: min. length=20, max. length=125, 27043027 kept reads (4967 discarded)\n",
      "[2025-03-15 16:45:21] Building transcriptome data files /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_13_S111_L005_R1_001/tmp/Arabidopsis_thaliana.TAIR10.54\n",
      "[2025-03-15 16:45:26] Building Bowtie index from Arabidopsis_thaliana.TAIR10.54.fa\n",
      "\t left reads: min. length=20, max. length=125, 28045007 kept reads (4487 discarded)\n",
      "[2025-03-15 16:45:28] Building transcriptome data files /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane5_FastqFiles/trimmed_Long_28_S114_L005_R1_001/tmp/Arabidopsis_thaliana.TAIR10.54\n",
      "[2025-03-15 16:45:33] Building Bowtie index from Arabidopsis_thaliana.TAIR10.54.fa\n",
      "[2025-03-15 16:47:00] Mapping left_kept_reads to transcriptome Arabidopsis_thaliana.TAIR10.54 with Bowtie2 \n",
      "[2025-03-15 16:47:17] Mapping left_kept_reads to transcriptome Arabidopsis_thaliana.TAIR10.54 with Bowtie2 \n",
      "[2025-03-15 16:47:37] Mapping left_kept_reads to transcriptome Arabidopsis_thaliana.TAIR10.54 with Bowtie2 \n",
      "[2025-03-15 16:49:26] Mapping left_kept_reads to transcriptome Arabidopsis_thaliana.TAIR10.54 with Bowtie2 \n",
      "[2025-03-15 16:49:33] Mapping left_kept_reads to transcriptome Arabidopsis_thaliana.TAIR10.54 with Bowtie2 \n",
      "[2025-03-15 16:50:25] Mapping left_kept_reads to transcriptome Arabidopsis_thaliana.TAIR10.54 with Bowtie2 \n",
      "[2025-03-15 16:50:59] Mapping left_kept_reads to transcriptome Arabidopsis_thaliana.TAIR10.54 with Bowtie2 \n",
      "[2025-03-15 16:51:10] Mapping left_kept_reads to transcriptome Arabidopsis_thaliana.TAIR10.54 with Bowtie2 \n",
      "[2025-03-15 16:51:14] Resuming TopHat pipeline with unmapped reads\n",
      "[2025-03-15 16:51:14] Mapping left_kept_reads.m2g_um to genome TAIR10_index with Bowtie2 \n",
      "[2025-03-15 16:52:18] Mapping left_kept_reads.m2g_um_seg1 to genome TAIR10_index with Bowtie2 (1/5)\n",
      "[2025-03-15 16:52:24] Mapping left_kept_reads.m2g_um_seg2 to genome TAIR10_index with Bowtie2 (2/5)\n",
      "[2025-03-15 16:52:32] Mapping left_kept_reads.m2g_um_seg3 to genome TAIR10_index with Bowtie2 (3/5)\n",
      "[2025-03-15 16:52:40] Mapping left_kept_reads.m2g_um_seg4 to genome TAIR10_index with Bowtie2 (4/5)\n",
      "[2025-03-15 16:52:45] Resuming TopHat pipeline with unmapped reads\n",
      "[2025-03-15 16:52:45] Mapping left_kept_reads.m2g_um to genome TAIR10_index with Bowtie2 \n",
      "[2025-03-15 16:52:49] Mapping left_kept_reads.m2g_um_seg5 to genome TAIR10_index with Bowtie2 (5/5)\n",
      "[2025-03-15 16:52:55] Searching for junctions via segment mapping\n",
      "[2025-03-15 16:53:15] Retrieving sequences for splices\n",
      "[2025-03-15 16:53:19] Indexing splices\n",
      "Building a SMALL index\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_3_S101_L004_R1_001/tmp/segment_juncs.3.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_3_S101_L004_R1_001/tmp/segment_juncs.3.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_3_S101_L004_R1_001/tmp/segment_juncs.4.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_3_S101_L004_R1_001/tmp/segment_juncs.4.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_3_S101_L004_R1_001/tmp/segment_juncs.1.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_3_S101_L004_R1_001/tmp/segment_juncs.1.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_3_S101_L004_R1_001/tmp/segment_juncs.2.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_3_S101_L004_R1_001/tmp/segment_juncs.2.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_3_S101_L004_R1_001/tmp/segment_juncs.rev.1.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_3_S101_L004_R1_001/tmp/segment_juncs.rev.1.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_3_S101_L004_R1_001/tmp/segment_juncs.rev.2.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_3_S101_L004_R1_001/tmp/segment_juncs.rev.2.bt2\n",
      "[2025-03-15 16:53:28] Mapping left_kept_reads.m2g_um_seg1 to genome segment_juncs with Bowtie2 (1/5)\n",
      "[2025-03-15 16:53:31] Mapping left_kept_reads.m2g_um_seg2 to genome segment_juncs with Bowtie2 (2/5)\n",
      "[2025-03-15 16:53:32] Resuming TopHat pipeline with unmapped reads\n",
      "[2025-03-15 16:53:32] Mapping left_kept_reads.m2g_um to genome TAIR10_index with Bowtie2 \n",
      "[2025-03-15 16:53:36] Mapping left_kept_reads.m2g_um_seg3 to genome segment_juncs with Bowtie2 (3/5)\n",
      "[2025-03-15 16:53:40] Mapping left_kept_reads.m2g_um_seg4 to genome segment_juncs with Bowtie2 (4/5)\n",
      "[2025-03-15 16:53:44] Mapping left_kept_reads.m2g_um_seg5 to genome segment_juncs with Bowtie2 (5/5)\n",
      "[2025-03-15 16:53:47] Joining segment hits\n",
      "[2025-03-15 16:53:55] Reporting output tracks\n",
      "[2025-03-15 16:54:24] Mapping left_kept_reads.m2g_um_seg1 to genome TAIR10_index with Bowtie2 (1/5)\n",
      "[2025-03-15 16:54:34] Mapping left_kept_reads.m2g_um_seg2 to genome TAIR10_index with Bowtie2 (2/5)\n",
      "[2025-03-15 16:54:45] Mapping left_kept_reads.m2g_um_seg3 to genome TAIR10_index with Bowtie2 (3/5)\n",
      "[2025-03-15 16:54:56] Mapping left_kept_reads.m2g_um_seg4 to genome TAIR10_index with Bowtie2 (4/5)\n",
      "[2025-03-15 16:55:00] Mapping left_kept_reads.m2g_um_seg1 to genome TAIR10_index with Bowtie2 (1/5)\n",
      "[2025-03-15 16:55:09] Mapping left_kept_reads.m2g_um_seg2 to genome TAIR10_index with Bowtie2 (2/5)\n",
      "[2025-03-15 16:55:09] Mapping left_kept_reads.m2g_um_seg5 to genome TAIR10_index with Bowtie2 (5/5)\n",
      "[2025-03-15 16:55:17] Searching for junctions via segment mapping\n",
      "[2025-03-15 16:55:18] Mapping left_kept_reads.m2g_um_seg3 to genome TAIR10_index with Bowtie2 (3/5)\n",
      "[2025-03-15 16:55:28] Mapping left_kept_reads.m2g_um_seg4 to genome TAIR10_index with Bowtie2 (4/5)\n",
      "[2025-03-15 16:55:39] Retrieving sequences for splices\n",
      "[2025-03-15 16:55:42] Mapping left_kept_reads.m2g_um_seg5 to genome TAIR10_index with Bowtie2 (5/5)\n",
      "[2025-03-15 16:55:42] Indexing splices\n",
      "Building a SMALL index\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_11_S109_L004_R1_001/tmp/segment_juncs.3.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_11_S109_L004_R1_001/tmp/segment_juncs.3.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_11_S109_L004_R1_001/tmp/segment_juncs.4.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_11_S109_L004_R1_001/tmp/segment_juncs.4.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_11_S109_L004_R1_001/tmp/segment_juncs.1.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_11_S109_L004_R1_001/tmp/segment_juncs.1.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_11_S109_L004_R1_001/tmp/segment_juncs.2.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_11_S109_L004_R1_001/tmp/segment_juncs.2.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_11_S109_L004_R1_001/tmp/segment_juncs.rev.1.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_11_S109_L004_R1_001/tmp/segment_juncs.rev.1.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_11_S109_L004_R1_001/tmp/segment_juncs.rev.2.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_11_S109_L004_R1_001/tmp/segment_juncs.rev.2.bt2\n",
      "[2025-03-15 16:55:50] Searching for junctions via segment mapping\n",
      "[2025-03-15 16:55:51] Mapping left_kept_reads.m2g_um_seg1 to genome segment_juncs with Bowtie2 (1/5)\n",
      "[2025-03-15 16:55:56] Mapping left_kept_reads.m2g_um_seg2 to genome segment_juncs with Bowtie2 (2/5)\n",
      "[2025-03-15 16:56:03] Mapping left_kept_reads.m2g_um_seg3 to genome segment_juncs with Bowtie2 (3/5)\n",
      "[2025-03-15 16:56:08] Mapping left_kept_reads.m2g_um_seg4 to genome segment_juncs with Bowtie2 (4/5)\n",
      "[2025-03-15 16:56:13] Retrieving sequences for splices\n",
      "[2025-03-15 16:56:14] Mapping left_kept_reads.m2g_um_seg5 to genome segment_juncs with Bowtie2 (5/5)\n",
      "[2025-03-15 16:56:16] Indexing splices\n",
      "Building a SMALL index\n",
      "[2025-03-15 16:56:18] Joining segment hits\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_2_S100_L004_R1_001/tmp/segment_juncs.3.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_2_S100_L004_R1_001/tmp/segment_juncs.3.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_2_S100_L004_R1_001/tmp/segment_juncs.4.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_2_S100_L004_R1_001/tmp/segment_juncs.4.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_2_S100_L004_R1_001/tmp/segment_juncs.1.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_2_S100_L004_R1_001/tmp/segment_juncs.1.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_2_S100_L004_R1_001/tmp/segment_juncs.2.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_2_S100_L004_R1_001/tmp/segment_juncs.2.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_2_S100_L004_R1_001/tmp/segment_juncs.rev.1.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_2_S100_L004_R1_001/tmp/segment_juncs.rev.1.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_2_S100_L004_R1_001/tmp/segment_juncs.rev.2.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_2_S100_L004_R1_001/tmp/segment_juncs.rev.2.bt2\n",
      "[2025-03-15 16:56:25] Mapping left_kept_reads.m2g_um_seg1 to genome segment_juncs with Bowtie2 (1/5)\n",
      "[2025-03-15 16:56:29] Reporting output tracks\n",
      "[2025-03-15 16:56:29] Mapping left_kept_reads.m2g_um_seg2 to genome segment_juncs with Bowtie2 (2/5)\n",
      "[2025-03-15 16:56:36] Mapping left_kept_reads.m2g_um_seg3 to genome segment_juncs with Bowtie2 (3/5)\n",
      "[2025-03-15 16:56:41] Mapping left_kept_reads.m2g_um_seg4 to genome segment_juncs with Bowtie2 (4/5)\n",
      "[2025-03-15 16:56:46] Mapping left_kept_reads.m2g_um_seg5 to genome segment_juncs with Bowtie2 (5/5)\n",
      "[2025-03-15 16:56:50] Joining segment hits\n",
      "[2025-03-15 16:57:00] Reporting output tracks\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Completed trimmed_Long_3_S101_L004_R1_001.\n",
      "trimmed_Long_10_S108_L004_R1_001 now being aligned\n",
      "Running TopHat2 for trimmed_Long_10_S108_L004_R1_001...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "-----------------------------------------------\n",
      "[2025-03-15 16:57:07] A summary of the alignment counts can be found in /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_3_S101_L004_R1_001/align_summary.txt\n",
      "[2025-03-15 16:57:07] Run complete: 00:17:16 elapsed\n",
      "\n",
      "[2025-03-15 16:57:07] Beginning TopHat run (v2.1.1)\n",
      "-----------------------------------------------\n",
      "[2025-03-15 16:57:07] Checking for Bowtie\n",
      "\t\t  Bowtie version:\t 2.5.4.0\n",
      "[2025-03-15 16:57:07] Checking for Bowtie index files (genome)..\n",
      "[2025-03-15 16:57:07] Checking for reference FASTA file\n",
      "[2025-03-15 16:57:07] Generating SAM header for /home/temccrac/Programs/data/genomes/TAIR10_index\n",
      "[2025-03-15 16:57:07] Reading known junctions from GTF file\n",
      "[2025-03-15 16:57:11] Preparing reads\n",
      "\t left reads: min. length=20, max. length=125, 10465739 kept reads (1445 discarded)\n",
      "[2025-03-15 16:59:11] Building transcriptome data files /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_10_S108_L004_R1_001/tmp/Arabidopsis_thaliana.TAIR10.54\n",
      "[2025-03-15 16:59:16] Building Bowtie index from Arabidopsis_thaliana.TAIR10.54.fa\n",
      "[2025-03-15 16:59:54] Resuming TopHat pipeline with unmapped reads\n",
      "[2025-03-15 16:59:54] Mapping left_kept_reads.m2g_um to genome TAIR10_index with Bowtie2 \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Completed trimmed_Long_11_S109_L004_R1_001.\n",
      "trimmed_Long_9_S107_L004_R1_001 now being aligned\n",
      "Running TopHat2 for trimmed_Long_9_S107_L004_R1_001...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "-----------------------------------------------\n",
      "[2025-03-15 17:00:07] A summary of the alignment counts can be found in /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_11_S109_L004_R1_001/align_summary.txt\n",
      "[2025-03-15 17:00:07] Run complete: 00:20:16 elapsed\n",
      "\n",
      "[2025-03-15 17:00:07] Beginning TopHat run (v2.1.1)\n",
      "-----------------------------------------------\n",
      "[2025-03-15 17:00:07] Checking for Bowtie\n",
      "\t\t  Bowtie version:\t 2.5.4.0\n",
      "[2025-03-15 17:00:07] Checking for Bowtie index files (genome)..\n",
      "[2025-03-15 17:00:07] Checking for reference FASTA file\n",
      "[2025-03-15 17:00:07] Generating SAM header for /home/temccrac/Programs/data/genomes/TAIR10_index\n",
      "[2025-03-15 17:00:07] Reading known junctions from GTF file\n",
      "[2025-03-15 17:00:11] Preparing reads\n",
      "[2025-03-15 17:00:52] Resuming TopHat pipeline with unmapped reads\n",
      "[2025-03-15 17:00:52] Mapping left_kept_reads.m2g_um to genome TAIR10_index with Bowtie2 \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Completed trimmed_Long_2_S100_L004_R1_001.\n",
      "trimmed_Long_4_S102_L004_R1_001 now being aligned\n",
      "Running TopHat2 for trimmed_Long_4_S102_L004_R1_001...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "-----------------------------------------------\n",
      "[2025-03-15 17:01:03] A summary of the alignment counts can be found in /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_2_S100_L004_R1_001/align_summary.txt\n",
      "[2025-03-15 17:01:03] Run complete: 00:21:12 elapsed\n",
      "\n",
      "[2025-03-15 17:01:03] Beginning TopHat run (v2.1.1)\n",
      "-----------------------------------------------\n",
      "[2025-03-15 17:01:03] Checking for Bowtie\n",
      "\t\t  Bowtie version:\t 2.5.4.0\n",
      "[2025-03-15 17:01:03] Checking for Bowtie index files (genome)..\n",
      "[2025-03-15 17:01:03] Checking for reference FASTA file\n",
      "[2025-03-15 17:01:03] Generating SAM header for /home/temccrac/Programs/data/genomes/TAIR10_index\n",
      "[2025-03-15 17:01:03] Reading known junctions from GTF file\n",
      "[2025-03-15 17:01:07] Preparing reads\n",
      "\t left reads: min. length=20, max. length=125, 11459220 kept reads (1728 discarded)\n",
      "[2025-03-15 17:02:22] Building transcriptome data files /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_9_S107_L004_R1_001/tmp/Arabidopsis_thaliana.TAIR10.54\n",
      "[2025-03-15 17:02:24] Mapping left_kept_reads.m2g_um_seg1 to genome TAIR10_index with Bowtie2 (1/5)\n",
      "[2025-03-15 17:02:27] Building Bowtie index from Arabidopsis_thaliana.TAIR10.54.fa\n",
      "[2025-03-15 17:02:39] Mapping left_kept_reads.m2g_um_seg2 to genome TAIR10_index with Bowtie2 (2/5)\n",
      "[2025-03-15 17:02:56] Mapping left_kept_reads.m2g_um_seg3 to genome TAIR10_index with Bowtie2 (3/5)\n",
      "[2025-03-15 17:03:12] Mapping left_kept_reads.m2g_um_seg4 to genome TAIR10_index with Bowtie2 (4/5)\n",
      "[2025-03-15 17:03:33] Mapping left_kept_reads.m2g_um_seg5 to genome TAIR10_index with Bowtie2 (5/5)\n",
      "[2025-03-15 17:03:45] Searching for junctions via segment mapping\n",
      "[2025-03-15 17:03:46] Mapping left_kept_reads.m2g_um_seg1 to genome TAIR10_index with Bowtie2 (1/5)\n",
      "[2025-03-15 17:04:01] Mapping left_kept_reads.m2g_um_seg2 to genome TAIR10_index with Bowtie2 (2/5)\n",
      "\t left reads: min. length=20, max. length=125, 15734298 kept reads (1418 discarded)\n",
      "[2025-03-15 17:04:06] Building transcriptome data files /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_4_S102_L004_R1_001/tmp/Arabidopsis_thaliana.TAIR10.54\n",
      "[2025-03-15 17:04:11] Building Bowtie index from Arabidopsis_thaliana.TAIR10.54.fa\n",
      "[2025-03-15 17:04:19] Mapping left_kept_reads.m2g_um_seg3 to genome TAIR10_index with Bowtie2 (3/5)\n",
      "[2025-03-15 17:04:23] Retrieving sequences for splices\n",
      "[2025-03-15 17:04:27] Indexing splices\n",
      "Building a SMALL index\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_7_S105_L004_R1_001/tmp/segment_juncs.3.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_7_S105_L004_R1_001/tmp/segment_juncs.3.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_7_S105_L004_R1_001/tmp/segment_juncs.4.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_7_S105_L004_R1_001/tmp/segment_juncs.4.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_7_S105_L004_R1_001/tmp/segment_juncs.1.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_7_S105_L004_R1_001/tmp/segment_juncs.1.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_7_S105_L004_R1_001/tmp/segment_juncs.2.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_7_S105_L004_R1_001/tmp/segment_juncs.2.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_7_S105_L004_R1_001/tmp/segment_juncs.rev.1.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_7_S105_L004_R1_001/tmp/segment_juncs.rev.1.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_7_S105_L004_R1_001/tmp/segment_juncs.rev.2.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_7_S105_L004_R1_001/tmp/segment_juncs.rev.2.bt2\n",
      "[2025-03-15 17:04:36] Mapping left_kept_reads.m2g_um_seg4 to genome TAIR10_index with Bowtie2 (4/5)\n",
      "[2025-03-15 17:04:36] Mapping left_kept_reads.m2g_um_seg1 to genome segment_juncs with Bowtie2 (1/5)\n",
      "[2025-03-15 17:04:44] Mapping left_kept_reads.m2g_um_seg2 to genome segment_juncs with Bowtie2 (2/5)\n",
      "[2025-03-15 17:04:51] Resuming TopHat pipeline with unmapped reads\n",
      "[2025-03-15 17:04:51] Mapping left_kept_reads.m2g_um to genome TAIR10_index with Bowtie2 \n",
      "[2025-03-15 17:04:54] Mapping left_kept_reads.m2g_um_seg3 to genome segment_juncs with Bowtie2 (3/5)\n",
      "[2025-03-15 17:04:55] Mapping left_kept_reads to transcriptome Arabidopsis_thaliana.TAIR10.54 with Bowtie2 \n",
      "[2025-03-15 17:04:59] Mapping left_kept_reads.m2g_um_seg5 to genome TAIR10_index with Bowtie2 (5/5)\n",
      "[2025-03-15 17:05:02] Mapping left_kept_reads.m2g_um_seg4 to genome segment_juncs with Bowtie2 (4/5)\n",
      "[2025-03-15 17:05:10] Resuming TopHat pipeline with unmapped reads\n",
      "[2025-03-15 17:05:10] Mapping left_kept_reads.m2g_um to genome TAIR10_index with Bowtie2 \n",
      "[2025-03-15 17:05:10] Mapping left_kept_reads.m2g_um_seg5 to genome segment_juncs with Bowtie2 (5/5)\n",
      "[2025-03-15 17:05:12] Searching for junctions via segment mapping\n",
      "[2025-03-15 17:05:17] Joining segment hits\n",
      "[2025-03-15 17:05:31] Reporting output tracks\n",
      "[2025-03-15 17:05:35] Resuming TopHat pipeline with unmapped reads\n",
      "[2025-03-15 17:05:35] Mapping left_kept_reads.m2g_um to genome TAIR10_index with Bowtie2 \n",
      "[2025-03-15 17:05:52] Retrieving sequences for splices\n",
      "[2025-03-15 17:05:56] Indexing splices\n",
      "Building a SMALL index\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_1_S99_L004_R1_001/tmp/segment_juncs.3.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_1_S99_L004_R1_001/tmp/segment_juncs.3.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_1_S99_L004_R1_001/tmp/segment_juncs.4.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_1_S99_L004_R1_001/tmp/segment_juncs.4.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_1_S99_L004_R1_001/tmp/segment_juncs.1.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_1_S99_L004_R1_001/tmp/segment_juncs.1.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_1_S99_L004_R1_001/tmp/segment_juncs.2.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_1_S99_L004_R1_001/tmp/segment_juncs.2.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_1_S99_L004_R1_001/tmp/segment_juncs.rev.1.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_1_S99_L004_R1_001/tmp/segment_juncs.rev.1.bt2\n",
      "Renaming /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_1_S99_L004_R1_001/tmp/segment_juncs.rev.2.bt2.tmp to /home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_1_S99_L004_R1_001/tmp/segment_juncs.rev.2.bt2\n",
      "[2025-03-15 17:06:06] Mapping left_kept_reads.m2g_um_seg1 to genome segment_juncs with Bowtie2 (1/5)\n",
      "[2025-03-15 17:06:14] Mapping left_kept_reads.m2g_um_seg2 to genome segment_juncs with Bowtie2 (2/5)\n",
      "[2025-03-15 17:06:25] Mapping left_kept_reads.m2g_um_seg3 to genome segment_juncs with Bowtie2 (3/5)\n",
      "[2025-03-15 17:06:34] Mapping left_kept_reads.m2g_um_seg4 to genome segment_juncs with Bowtie2 (4/5)\n",
      "[2025-03-15 17:06:43] Mapping left_kept_reads.m2g_um_seg5 to genome segment_juncs with Bowtie2 (5/5)\n",
      "[2025-03-15 17:06:50] Joining segment hits\n",
      "[2025-03-15 17:07:06] Reporting output tracks\n",
      "[2025-03-15 17:08:02] Mapping left_kept_reads to transcriptome Arabidopsis_thaliana.TAIR10.54 with Bowtie2 \n",
      "[2025-03-15 17:08:42] Mapping left_kept_reads.m2g_um_seg1 to genome TAIR10_index with Bowtie2 (1/5)\n",
      "[2025-03-15 17:08:59] Mapping left_kept_reads.m2g_um_seg1 to genome TAIR10_index with Bowtie2 (1/5)\n",
      "[2025-03-15 17:08:59] Mapping left_kept_reads.m2g_um_seg2 to genome TAIR10_index with Bowtie2 (2/5)\n",
      "[2025-03-15 17:09:11] Mapping left_kept_reads.m2g_um_seg1 to genome TAIR10_index with Bowtie2 (1/5)\n",
      "[2025-03-15 17:09:19] Mapping left_kept_reads.m2g_um_seg2 to genome TAIR10_index with Bowtie2 (2/5)\n",
      "[2025-03-15 17:09:19] Mapping left_kept_reads.m2g_um_seg3 to genome TAIR10_index with Bowtie2 (3/5)\n",
      "[2025-03-15 17:09:28] Mapping left_kept_reads.m2g_um_seg2 to genome TAIR10_index with Bowtie2 (2/5)\n",
      "[2025-03-15 17:09:39] Mapping left_kept_reads.m2g_um_seg4 to genome TAIR10_index with Bowtie2 (4/5)\n",
      "[2025-03-15 17:09:41] Mapping left_kept_reads.m2g_um_seg3 to genome TAIR10_index with Bowtie2 (3/5)\n",
      "[2025-03-15 17:09:46] Mapping left_kept_reads to transcriptome Arabidopsis_thaliana.TAIR10.54 with Bowtie2 \n",
      "[2025-03-15 17:09:49] Mapping left_kept_reads.m2g_um_seg3 to genome TAIR10_index with Bowtie2 (3/5)\n",
      "[2025-03-15 17:10:03] Mapping left_kept_reads.m2g_um_seg4 to genome TAIR10_index with Bowtie2 (4/5)\n",
      "[2025-03-15 17:10:06] Mapping left_kept_reads.m2g_um_seg5 to genome TAIR10_index with Bowtie2 (5/5)\n"
     ]
    }
   ],
   "source": [
    "# Parallel processing for sequence alignment\n",
    "import os\n",
    "import subprocess\n",
    "from multiprocessing import Pool, cpu_count\n",
    "\n",
    "# Define paths\n",
    "tophat2_path = \"/home/temccrac/Programs/environments/tophat-2.1.1.Linux_x86_64/tophat\"\n",
    "bowtie2_index = \"/home/temccrac/Programs/data/genomes/TAIR10_index\" # path to index prefix\n",
    "gtf_file = \"/home/temccrac/Programs/data/genomes/Arabidopsis_thaliana.TAIR10.54.gtf\"\n",
    "\n",
    "# Function to run TopHat2 for a single file\n",
    "def run_tophat(fastq_file_path):\n",
    "    dir_name_detected = None  # Initialize\n",
    "    # Loop through each directory name to find a match\n",
    "    for dir_name in dir_names:\n",
    "        if f\"/{dir_name}/\" in fastq_file_path:  # Ensures exact match as directory\n",
    "            dir_name_detected = dir_name\n",
    "            break  # Stop searching after first match\n",
    "\n",
    "    # Only print an error if no match was found after the loop finishes\n",
    "    if dir_name_detected is None:\n",
    "        print(f\"Error: No matching directory found for {fastq_file_path}\")\n",
    "        return  # Exit function if no match is found\n",
    "\n",
    "    sample_name = os.path.basename(fastq_file_path).replace(\".fastq.gz\", \"\")\n",
    "    print(sample_name, 'now being aligned')\n",
    "    output_dir_path = os.path.join(sequence_path, dir_name_detected, sample_name)\n",
    "    \n",
    "    # Construct TopHat2 command\n",
    "    command = [\n",
    "        \"python2\", tophat2_path,  # Ensures Python2 is used\n",
    "        \"-o\", output_dir_path,\n",
    "        \"-G\", gtf_file,\n",
    "        \"-p\", \"4\",  # Number of threads per job\n",
    "        bowtie2_index,\n",
    "        fastq_file_path\n",
    "    ]\n",
    "    \n",
    "    # Run the command\n",
    "    print(f\"Running TopHat2 for {sample_name}...\")\n",
    "    subprocess.run(command, check=True)\n",
    "    print(f\"Completed {sample_name}.\")\n",
    "\n",
    "# Number of parallel processes\n",
    "num_cores = 8  # Use as many cores as available\n",
    "\n",
    "# Run in parallel\n",
    "with Pool(processes=num_cores) as pool:\n",
    "    pool.map(run_tophat, fastq_files)\n",
    "\n",
    "print(\"All alignments completed!\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Reccomended by ChatGPT for concatinating the output bam files from alignment into a matrix\n",
    "# Do I need to move all bam files to one folder, hehe?\n",
    "featureCounts -T 8 -a Arabidopsis_thaliana.TAIR10.54.gtf -o gene_counts.txt *.bam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Lane4_FastqFiles',\n",
       " 'Lane5_FastqFiles',\n",
       " 'Lane6_FastqFiles',\n",
       " 'Lane7_FastqFiles',\n",
       " '20171212_GSL_105B_Long']"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dir_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/temccrac/Programs/data/cellular_clarity/adrabidopsis_data/Lane4_FastqFiles/trimmed_Long_3_S101_L004_R1_001.fastq.gz'"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trimmed_sequence_dictionary[dir_names[0]][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Command for tophat\n",
    "tophat2 -o /path/to/output \\\n",
    "  -G ~/genomes/TAIR10/Arabidopsis_thaliana.TAIR10.54.gtf \\\n",
    "  -p 4 \\\n",
    "  ~/genomes/TAIR10/TAIR10_index \\\n",
    "  /path/to/sample.fastq.gz"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "transcriptomics",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
